pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -131,6 +191,48 @@ func (r *ReconcileUserSignup) Reconcile(request reconcile.Request) (reconcile.Re
 		return reconcile.Result{}, err
 	}
 
+	// Is the user banned? To determine this we query the BannedUser resource for any matching entries.  The query
+	// is based on the user's emailHash value - if there is a match, and the e-mail addresses are equal, then the
+	// user is banned.
+	banned := false
+
+	// Lookup the user email annotation
+	if emailLbl, exists := instance.Annotations[toolchainv1alpha1.UserSignupUserEmailAnnotationKey]; exists {
+
+		// Lookup the email hash label
+		if emailHashLbl, exists := instance.Labels[toolchainv1alpha1.UserSignupUserEmailHashAnnotationKey]; exists {
+
+			labels := map[string]string{toolchainv1alpha1.BannedUserEmailHashLabelKey: emailHashLbl}
+			opts := client.MatchingLabels(labels)
+			bannedUserList := &toolchainv1alpha1.BannedUserList{}
+
+			// Query BannedUser for resources that match the same email hash
+			if err = r.client.List(context.TODO(), bannedUserList, opts); err != nil {
+				return reconcile.Result{}, r.wrapErrorWithStatusUpdate(reqLogger, instance, r.setStatusFailedToReadBannedUsers, err, "Failed to query BannedUsers")
+			}
+
+			// One last check to confirm that the e-mail addresses match also (in case of the infinitesimal chance of a hash collision)
+			for _, bannedUser := range bannedUserList.Items {
+				if bannedUser.Spec.Email == emailLbl {
+					banned = true
+					break
+				}
+			}
+
+			hashIsValid, err := validateEmailHash(emailLbl, emailHashLbl)
+			if err != nil {
+				return reconcile.Result{}, r.updateStatus(reqLogger, instance, r.setStatusFailedToValidateEmailHash)
+			} else if !hashIsValid {
+				return reconcile.Result{}, r.updateStatus(reqLogger, instance, r.setStatusInvalidEmailHash)
```

Comment:
Same here:
```suggestion
				err := errors.New("invalid hash") 
				return reconcile.Result{}, r.wrapErrorWithStatusUpdate(reqLogger, instance, r.setStatusInvalidEmailHash, err, "")
```

Reply:
Similar to above - if we return an error here, the controller continually re-attempt to reconcile, with the identical result every time.  If the e-mail hash is invalid, then the current design requires user intervention to correct the hash value, and this intervention will re-trigger the reconciliation anyway.